tests/devices/test_default_qubit.py

```diff
@@ -1989,3 +1661,133 @@ def test_s_inverse():
 
         expected = np.array([1., -1.j]) / np.sqrt(2)
         assert np.allclose(dev.state, expected, atol=tol, rtol=0)
+
+@pytest.mark.parametrize("inverse", [True, False])
```

Comment:
These tests are new, they test the `_apply_operation` method

Reply:
amazing!